File: src/main/java/gg/cookingdom/controller/ImageController.java
package gg.cookingdom.controller;

import gg.cookingdom.service.image.ImagesService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/image")
@RequiredArgsConstructor
public class ImageController {

    private final ImagesService imagesService;

    @GetMapping(
            value = "/{imageName}",
            produces = {MediaType.IMAGE_JPEG_VALUE, MediaType.IMAGE_PNG_VALUE, MediaType.IMAGE_GIF_VALUE}
    )
    public byte[] getImage(@PathVariable("image-name") String imageName) {
        return imagesService.getImage(imageName);
    }
}

File: src/main/java/gg/cookingdom/controller/ImageController.kt
package gg.cookingdom.controller

import gg.cookingdom.service.image.ImageService
import org.springframework.cache.annotation.Cacheable
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/image")
class ImageController(
    private val imageService: ImageService
) {
    @GetMapping(
        value = ["/{imageName}"],
        produces = [MediaType.IMAGE_JPEG_VALUE, MediaType.IMAGE_PNG_VALUE, MediaType.IMAGE_GIF_VALUE]
    )
    fun getImagesByName(@PathVariable imageName: String): ByteArray {
        return imageService.getImage(imageName)
    }
}
